watertap/unit_models/translator_adm1_asm1.py

```diff
@@ -0,0 +1,463 @@
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality alkalinity equation",
+        )
+        def return_Salk(blk, t):
+            return (
+                blk.properties_out[t].alkalinity
+                == blk.properties_in[t].conc_mass_comp["S_IC"] / mw_c
+            )
+
+        self.zero_flow_components = Set(
+            initialize=["X_BH", "X_BA", "X_P", "S_O", "S_NO"]
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.zero_flow_components,
+            doc="Components with no flow equation",
+        )
+        def return_zero_flow_comp(blk, t, i):
+            return (
+                blk.properties_out[t].conc_mass_comp[i]
+                == 1e-6 * pyunits.kg / pyunits.m**3
+            )
+
+    def initialize_build(
+        self,
+        state_args_in=None,
+        state_args_out=None,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+    ):
+        """
+        This method calls the initialization method of the state blocks.
+
+        Keyword Arguments:
+            state_args_in : a dict of arguments to be passed to the inlet
+                property package (to provide an initial state for
+                initialization (see documentation of the specific
+                property package) (default = None).
+            state_args_out : a dict of arguments to be passed to the outlet
+                property package (to provide an initial state for
+                initialization (see documentation of the specific
+                property package) (default = None).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None, use
+                     default solver options)
+            solver : str indicating which solver to use during
+                     initialization (default = None, use default solver)
+
+        Returns:
+            None
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+
+        # Create solver
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Initialize state block
+        flags = self.properties_in.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_in,
+            hold_state=True,
+        )
+
+        self.properties_out.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_out,
+        )
+
+        if degrees_of_freedom(self) == 0:
+            with idaeslog.solver_log(init_log, idaeslog.DEBUG) as slc:
+                res = opt.solve(self, tee=slc.tee)
+
+            init_log.info("Initialization Complete {}.".format(idaeslog.condition(res)))
+        else:
+            init_log.warning(
```

Comment:
I guess this was just taken from Translator template. I wonder if this was an oversight in IDAES or not. Usually, we raise exceptions for something like this, and I am wondering if IDAES' Translator's `initialize_build` routine intentionally relaxes our typical restrictions for some reason. I am leaning towards this just being an oversight, but maybe this is a question for @andrewlee94